lib/Mapper.php

```diff
@@ -863,6 +885,156 @@ public function upsert(array $data, array $where)
     }
 
     /**
+     * Save related entities that have been assigned or loaded (only HasOne, HasMany and HasManyThrough relations)
+     * See saveBelongsToRelations.
+     *
+     * @param \Spot\EntityInterface $entity
+     * @param array $options
+     * @return mixed
+     * @throws \InvalidArgumentException
+     */
+    public function saveHasRelations(EntityInterface $entity, array $options = [])
```

Comment:
This is a huge method. I'd like to break this up a bit before merging it in. Can you put these individual relation cases on the relation classes themselves? That way, we could do something like `$relation->saveRelatedEntites([ .. params ])` and move most of the work back to the relation classes themselves.